pkg/systemd/terminal.html

```diff
@@ -23,7 +23,23 @@
     var term = null;
     var channel = null;
 
+    var current_user = cockpit.user;
+
+    var terminal_timer = null;
+    var shell_update_started = null;
     function start_terminal() {
+        if (shell_update_started === null)
+            shell_update_started = new Date();
+
+        /* if we don't have user information yet, retry soon
+         * abort wait and just use default shell if we exceed time limit
+         */
+        if ((current_user.user === null) && ((new Date()) - shell_update_started < 1000)) {
```

Comment:
> Just if ((!current_user) && ...

Here I can use that, since that will also catch empty user strings... I'll change it.